src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -6704,6 +6720,705 @@ namespace HeatBalanceSurfaceManager {
+        Real64 HMovInsul;  // "Convection" coefficient of movable insulation
+        Real64 MaxDelTemp; // Maximum change in surface temperature for any
+        //  opaque surface from one iteration to the next
+        static Array1D<Real64> TempInsOld; // Holds previous iteration's value for convergence check
+
+        int OtherSideSurfNum;     // Surface number index for other side of an interzone partition
+        static int MinIterations; // Minimum number of iterations for the inside heat balance
+        static int ErrCount(0);
+        Real64 Ueff; // 1 / effective R value between TDD:DOME and TDD:DIFFUSER
+
+        int ZoneEquipConfigNum;
+        int NodeNum;
+        Real64 SumSysMCp;  // Zone sum of air system MassFlowRate*Cp
+        Real64 SumSysMCpT; // Zone sum of air system MassFlowRate*Cp*T
+        Real64 MassFlowRate;
+        Real64 NodeTemp;
+        Real64 CpAir;
+        static Array1D<Real64> RefAirTemp; // reference air temperatures
+        static bool MyEnvrnFlag(true);
+        static int InsideSurfErrCount(0);
+        static int WarmupSurfTemp;
+        static int TimeStepInDay(0); // time step number
+
+        if (calcHeatBalanceInsideSurfCTFOnlyFirstTime) {
+            TempInsOld.allocate(TotSurfaces);
+            RefAirTemp.allocate(TotSurfaces);
+            MinIterations = 1;
+            if (DisplayAdvancedReportVariables) {
+                SetupOutputVariable("Surface Inside Face Heat Balance Calculation Iteration Count",
+                                    OutputProcessor::Unit::None,
+                                    InsideSurfIterations,
+                                    "ZONE",
+                                    "Sum",
+                                    "Simulation");
+            }
+            // Precompute whether CTF temperature limits will be needed
+            DataHeatBalSurface::Zone_has_mixed_HT_models.resize(NumOfZones + 1, false);
+            for (int iZone = 1; iZone <= NumOfZones; ++iZone) {
+                auto const &zone(Zone(iZone));
+                for (int iSurf = zone.SurfaceFirst, eSurf = zone.SurfaceLast; iSurf <= eSurf; ++iSurf) {
+                    auto const alg(Surface(iSurf).HeatTransferAlgorithm);
+                    if ((alg == HeatTransferModel_CondFD) || (alg == HeatTransferModel_HAMT) || (alg == HeatTransferModel_Kiva)) {
+                        DataHeatBalSurface::Zone_has_mixed_HT_models[iZone] = true;
+                        break;
+                    }
+                }
+            }
+            // Set up coefficient arrays that never change - loop over non-window HT surfaces
+            for (int zoneNum = FirstZone; zoneNum <= LastZone; ++zoneNum) {
+                int const firstSurf = Zone(zoneNum).NonWindowSurfaceFirst;
+                int const lastSurf = Zone(zoneNum).NonWindowSurfaceLast;
+                for (int surfNum = firstSurf; surfNum <= lastSurf; ++surfNum) {
```

Comment:
Throughout this function, used nested zone then surface loops where possible in hopes that zone loop could become parallel someday. Too fragmented yet at this point, but moving in that direction.